go-client/client/tablehandle.go

```diff
@@ -51,6 +60,49 @@ func newTableHandle(client *Client, ticket *ticketpb2.Ticket, schema *arrow.Sche
 	}
 }
 
+// newTableHandle returns a new table handle given information about the table,
+// typically from a gRPC ExportedTableCreationResponse.
+// If WithTableLeaksAllowed was not provided when creating a client,
+// this will also attaches a finalizer to the TableHandle that will warn if the handle is leaked and free the table automatically.
```

Comment:
1) See earlier comment on what I thought we discussed as the feature.
2) Grammar.  attach